src/linquantarrays.jl

```diff
@@ -10,84 +13,164 @@ Base.size(QA::LinQuantArray) = size(QA.A)
+LinQuantUInt8Array(A::AbstractArray{T,N}, e::Option{Tuple}) where {T,N} = LinQuantization(UInt8,A,e)
+LinQuantUInt16Array(A::AbstractArray{T,N}, e::Option{Tuple}) where {T,N} = LinQuantization(UInt16,A,e)
+LinQuantUInt24Array(A::AbstractArray{T,N}, e::Option{Tuple}) where {T,N} = LinQuantization(UInt24,A,e)
+LinQuantUInt32Array(A::AbstractArray{T,N}, e::Option{Tuple}) where {T,N} = LinQuantization(UInt32,A,e)
+
+# define for signed integers of  8, 16, 24 and 32 bit
+LinQuantInt8Array(A::AbstractArray{T,N}, e::Option{Tuple}) where {T,N} = LinQuantization(Int8,A,e)
+LinQuantInt16Array(A::AbstractArray{T,N}, e::Option{Tuple}) where {T,N} = LinQuantization(Int16,A,e)
+LinQuantInt24Array(A::AbstractArray{T,N}, e::Option{Tuple}) where {T,N} = LinQuantization(Int24,A,e)
+LinQuantInt32Array(A::AbstractArray{T,N}, e::Option{Tuple}) where {T,N} = LinQuantization(Int32,A,e)
+
+
 """De-quantise a LinQuantArray into floats."""
-function Base.Array{T}(n::Integer,Q::LinQuantArray) where {T<:AbstractFloat}
-    Qmin = Q.min                # min as Float64
-    Qmax = Q.max                # max as Float64
-    Δ = (Qmax-Qmin)/(2^n-1)     # linear spacing
+function Base.Array{U}(n::Integer, Q::LinQuantArray) where {U<:AbstractFloat}
+    Qmin = Q.min                     # min of original Array as Float64
+    Qmax = Q.max                     # max of original Array as Float64
+    Tmin = Float64(typemin(eltype(Q)))  # min representable in type as Float64
+    Tmax = Float64(typemax(eltype(Q)))  # max representable in type as Float64
+    Δ = (Qmax-Qmin)/(Tmax-Tmin)          # linear spacing
 
-    A = similar(Q,T)
+    A = similar(Q,U)
 
     @inbounds for i in eachindex(A)
-        # convert Q[i]::UInt to Float64 via *
+        # convert Q[i]::Integer to Float64 via *
         # then to T through =
-        A[i] = Qmin + Q[i]*Δ
+        A[i] = Qmin + (Q[i] - Tmin)*Δ
     end
 
     return A
 end
 
-# define default conversions for 8, 16, 24 and 32 bit
+# define default conversions for unsigned 8, 16, 24 and 32 bit
 Base.Array{T}(Q::LinQuantArray{UInt8,N}) where {T,N} = Array{T}(8,Q)
 Base.Array{T}(Q::LinQuantArray{UInt16,N}) where {T,N} = Array{T}(16,Q)
 Base.Array{T}(Q::LinQuantArray{UInt24,N}) where {T,N} = Array{T}(24,Q)
 Base.Array{T}(Q::LinQuantArray{UInt32,N}) where {T,N} = Array{T}(32,Q)
 
-Base.Array(Q::LinQuantArray{UInt8,N}) where N = Array{Float32}(8,Q)
-Base.Array(Q::LinQuantArray{UInt16,N}) where N = Array{Float32}(16,Q)
-Base.Array(Q::LinQuantArray{UInt24,N}) where N = Array{Float32}(24,Q)
-Base.Array(Q::LinQuantArray{UInt32,N}) where N = Array{Float64}(32,Q)
+# define default conversions for signed 8, 16, 24 and 32 bit
+Base.Array(Q::LinQuantArray{Int8,N}) where N = Array{Float32}(8,Q)
+Base.Array(Q::LinQuantArray{Int16,N}) where N = Array{Float32}(16,Q)
+Base.Array(Q::LinQuantArray{Int24,N}) where N = Array{Float32}(24,Q)
+Base.Array(Q::LinQuantArray{Int32,N}) where N = Array{Float64}(32,Q)
```

Comment:
Once we take the `n` from the `eltype` in `Q` then these would only be needed to set a default floating-point type if not provided I guess!